src/nprintml/label_aggregation.py

```diff
@@ -0,0 +1,233 @@
+"""
+/*
+  * Copyright 2020 nPrint
+  * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+  * use this file except in compliance with the License. You may obtain a copy
+  * of the License at https://www.apache.org/licenses/LICENSE-2.0
+*/
+
+This module contains code for training, evaluating, and representing the results
+of AutoML training on nPrints. In general, it takes a dataframe generated from the
+label aggregation step of the nprintML pipeline. It can be used on its own
+as long as it receives a dataframe where each row of the dataframe contains
+a single sample and the labels for the samples are contained in a 'label' column
+in the dataframe
+
+"""
+
+import pathlib
+from abc import ABC, abstractmethod
+
+import numpy as np
+import pandas as pd
+
+
+class LabelAggregator(ABC):
```

Comment:
It seems apt to have an ABC for the label aggregator as much of it is the same. In general you always (at some point) want to 
* attach labels
* compress the nPrints
* load nPrints
* load labels
* get flattened (or multiplied?) columns

This was my first pass at it

Reply:
One more thought/suggestion on this – whether related to the above or not – rather than the usefully-named but specifically-named abstract method `generate_features`, the abstract method could be say `__call__`. This would not make much difference, (and here is largely a question of taste), but would merely simplify the interface slightly, down to:

    aggregator = ThisAggregator(npt_file, label_file)
    features = aggregator(compress=compress, sample_size=sample_size)

Reply:
My worry with removing it as an abstract class is that when we inevitably implement *more* aggregators we will likely want to switch back. With this method, if someone wants to add an aggregator (us or someone else), they have an interface to adhere to and it should be extremely clear what is happening. I do like the idea of the __call__ method though.